support/sys-mman.c

```diff
@@ -89,7 +89,7 @@ Mono_Posix_Syscall_msync (void *start, mph_size_t len, int flags)
 int
 Mono_Posix_Syscall_mlock (void *start, mph_size_t len)
 {
-#if !defined(HAVE_MINCORE)
+#if !defined(HAVE_MLOCK)
```

Comment:
Or __HAIKU__ like munlock?

Reply:
i'd recommend just testing for mlock/munlock, and that should cover Haiku